internal/warn_on_deprecated_api.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+// This module is browser compatible.
+
+// deno-lint-ignore no-explicit-any
+const { Deno } = globalThis as any;
+
+const ALREADY_WARNED_DEPRECATED = new Set<string>();
+const ENV_VAR_KEY = "DENO_NO_DEPRECATION_WARNINGS";
+const shouldDisableDeprecatedApiWarning =
+  (await Deno?.permissions.query({ name: "env", variable: ENV_VAR_KEY }))
```

Comment:
Top-level-await caused many issues in the past (The order of module executions became unpredictable with it). I think we should avoid it

I'd suggest we should use `querySync` for CLI and we can skip permission check in Deploy (env access is always possible in Deploy).

Reply:
Ah, good point. Fixed.